lib/config/hadeployconfigverify.go

```diff
@@ -0,0 +1,671 @@
+	if err != nil {
+		errorList.PushBack(err)
+	}
+
+	err = validateFQDN(externalOpensearchSettings.OpensearchDomainURL, "opensearch_domain_url")
+	if err != nil {
+		errorList.PushBack(err)
+	}
+
+	if c.IsExternalDbSelfManaged() {
+		err = validateRequiredString(externalOpensearchSettings.OpensearchRootCert, "opensearch_root_cert")
+		if err != nil {
+			errorList.PushBack(err)
+		}
+	}
+
+	err = validateRequiredString(externalOpensearchSettings.OpensearchUserPassword, "opensearch_user_password")
+	if err != nil {
+		errorList.PushBack(err)
+	}
+
+	err = validateRequiredString(externalOpensearchSettings.OpensearchUsername, "opensearch_username")
+	if err != nil {
+		errorList.PushBack(err)
+	}
+
+	return getSingleErrorFromList(errorList)
+}
+
+func (c *HaDeployConfig) verifyAwsExternalOsSettings(awsExternalOpensearchSettings *AwsExternalOsSettings) error {
+	errorList := list.New()
+
+	err := validateRequiredString(awsExternalOpensearchSettings.AwsOsSnapshotRoleArn, "aws_os_snapshot_role_arn")
+	if err != nil {
+		errorList.PushBack(err)
+	}
+
+	err = validateRequiredString(awsExternalOpensearchSettings.OsSnapshotUserAccessKeyID, "os_snapshot_user_access_key_id")
+	if err != nil {
+		errorList.PushBack(err)
+	}
+
+	err = validateRequiredString(awsExternalOpensearchSettings.OsSnapshotUserAccessKeySecret, "os_snapshot_user_access_key_secret")
+	if err != nil {
+		errorList.PushBack(err)
+	}
+
+	return getSingleErrorFromList(errorList)
+}
+
+func (c *HaDeployConfig) validateAwsSettings(aws *ConfigAwsSettings) error {
+	errorList := list.New()
+
+	if err := validateCommonAwsSettings(aws); err != nil {
+		errorList.PushBack(err)
+	}
+
+	if !aws.SetupManagedServices {
+		if err := validateAwsOsPgConfig(aws); err != nil {
+			errorList.PushBack(err)
+		}
+	} else {
+		// validate aws managed services
+		if err := validateAwsManagedServices(aws); err != nil {
+			errorList.PushBack(err)
+		}
+	}
+	return getSingleErrorFromList(errorList)
+}
+
+func (c *HaDeployConfig) IsValidHaDeployConfig() bool {
+	return c != nil && c.Architecture != nil
+}
+
+func (c *HaDeployConfig) IsExistingInfra() bool {
+	return c.Architecture != nil && c.Architecture.ExistingInfra != nil
+}
+
+func (c *HaDeployConfig) IsAws() bool {
+	return c.Architecture != nil && c.Architecture.Aws != nil
+}
+
+func (c *HaDeployConfig) IsExternalDb() bool {
+	return c.External != nil && c.External.Database != nil &&
+		(c.External.Database.Type == "aws" || c.External.Database.Type == "self-managed")
+}
+
+func (c *HaDeployConfig) IsAwsExternalOsConfigured() bool {
+	return c.External != nil && c.External.Database != nil &&
+		c.External.Database.OpenSearch != nil && c.External.Database.OpenSearch.Aws != nil &&
+		c.External.Database.OpenSearch.Aws.AwsOsSnapshotRoleArn != ""
+}
+
+func (c *HaDeployConfig) IsExternalDbSelfManaged() bool {
+	return c.IsExternalDb() && c.External.Database.Type == "self-managed"
```

Comment:
move `self-managed` to const if it used more than 1 place

Reply:
Okey.